javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +518,971 @@ object Character {
+      map.put("duployan", DUPLOYAN)
+      map.put("shorthandformatcontrols", SHORTHAND_FORMAT_CONTROLS)
+      map.put("byzantinemusicalsymbols", BYZANTINE_MUSICAL_SYMBOLS)
+      map.put("musicalsymbols", MUSICAL_SYMBOLS)
+      map.put("ancientgreekmusicalnotation", ANCIENT_GREEK_MUSICAL_NOTATION)
+      map.put("mayannumerals", MAYAN_NUMERALS)
+      map.put("taixuanjingsymbols", TAI_XUAN_JING_SYMBOLS)
+      map.put("countingrodnumerals", COUNTING_ROD_NUMERALS)
+      map.put("mathematicalalphanumericsymbols", MATHEMATICAL_ALPHANUMERIC_SYMBOLS)
+      map.put("glagoliticsupplement", GLAGOLITIC_SUPPLEMENT)
+      map.put("nyiakengpuachuehmong", NYIAKENG_PUACHUE_HMONG)
+      map.put("wancho", WANCHO)
+      map.put("mendekikakui", MENDE_KIKAKUI)
+      map.put("adlam", ADLAM)
+      map.put("indicsiyaqnumbers", INDIC_SIYAQ_NUMBERS)
+      map.put("ottomansiyaqnumbers", OTTOMAN_SIYAQ_NUMBERS)
+      map.put("arabicmathematicalalphabeticsymbols", ARABIC_MATHEMATICAL_ALPHABETIC_SYMBOLS)
+      map.put("mahjongtiles", MAHJONG_TILES)
+      map.put("dominotiles", DOMINO_TILES)
+      map.put("playingcards", PLAYING_CARDS)
+      map.put("enclosedalphanumericsupplement", ENCLOSED_ALPHANUMERIC_SUPPLEMENT)
+      map.put("enclosedideographicsupplement", ENCLOSED_IDEOGRAPHIC_SUPPLEMENT)
+      map.put("emoticons", EMOTICONS)
+      map.put("ornamentaldingbats", ORNAMENTAL_DINGBATS)
+      map.put("alchemicalsymbols", ANATOLIAN_HIEROGLYPHS)
+      map.put("geometricshapesextended", GEOMETRIC_SHAPES_EXTENDED)
+      map.put("supplementalarrowsc", SUPPLEMENTAL_ARROWS_C)
+      map.put("chesssymbols", CHESS_SYMBOLS)
+      map.put("cjkunifiedideographsextensionb", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B)
+      map.put("cjkunifiedideographsextensionc", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C)
+      map.put("cjkunifiedideographsextensiond", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D)
+      map.put("cjkunifiedideographsextensione", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_E)
+      map.put("cjkunifiedideographsextensionf", CJK_UNIFIED_IDEOGRAPHS_EXTENSION_F)
+      map.put("cjkcompatibilityideographssupplement", CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT)
+      map.put("tags", TAGS)
+      map.put("variationselectorssupplement", VARIATION_SELECTORS_SUPPLEMENT)
+      map.put("supplementaryprivateuseareaa", SUPPLEMENTARY_PRIVATE_USE_AREA_A)
+      map.put("supplementaryprivateuseareab", SUPPLEMENTARY_PRIVATE_USE_AREA_B)
+      map
+    }
+
+    // From: https://unicode.org/Public/UCD/latest/ucd/Blocks.txt
+    // Note:   When comparing block names, casing, whitespace, hyphens,
+    //         and underbars are ignored.
+    //         For example, "Latin Extended-A" and "latin extended a" are equivalent.
+    //         For more information on the comparison of property values,
+    //           see UAX #44: http://www.unicode.org/reports/tr44/
+    private def toNormalizedName(name: String): String = {
+      name.toLowerCase.replaceAll(raw"[\s\-_]", "")
```

Comment:
This may follow what the Unicode standard says about how to compare block names, but it does not match the specification of [`forName`](https://docs.oracle.com/javase/8/docs/api/java/lang/Character.UnicodeBlock.html#forName-java.lang.String-), which only accepts some specific forms. For example:
```scala
scala> Character.UnicodeBlock.forName("Basic Latin")
res4: Character.UnicodeBlock = BASIC_LATIN

scala> Character.UnicodeBlock.forName("Basic LatiN")
res6: Character.UnicodeBlock = BASIC_LATIN

scala> Character.UnicodeBlock.forName("Basic Lat iN")
java.lang.IllegalArgumentException
  at java.lang.Character$UnicodeBlock.forName(Character.java:3190)
  ... 28 elided

scala> Character.UnicodeBlock.forName("Basic Lat in")
java.lang.IllegalArgumentException
  at java.lang.Character$UnicodeBlock.forName(Character.java:3190)
  ... 28 elided

scala> Character.UnicodeBlock.forName("Arabic Extended-A")
res15: Character.UnicodeBlock = ARABIC_EXTENDED_A

scala> Character.UnicodeBlock.forName("Arabic_Extended-A")
java.lang.IllegalArgumentException
  at java.lang.Character$UnicodeBlock.forName(Character.java:3190)
  ... 28 elided

scala> Character.UnicodeBlock.forName("Arabic_Extended_A")
res17: Character.UnicodeBlock = ARABIC_EXTENDED_A

scala> Character.UnicodeBlock.forName("Arabic_ExtendedA")
java.lang.IllegalArgumentException
  at java.lang.Character$UnicodeBlock.forName(Character.java:3190)
  ... 28 elided

scala> Character.UnicodeBlock.forName("ArabicExtendedA")
java.lang.IllegalArgumentException
  at java.lang.Character$UnicodeBlock.forName(Character.java:3190)
  ... 28 elided

scala> Character.UnicodeBlock.forName("ArabicExtended-A")
res20: Character.UnicodeBlock = ARABIC_EXTENDED_A
```
Blindly removing all spaces, dashes and underscores is not correct. `toLowerCase` is fine, but after it needs to be an exact match for one of the three possible forms.